components/feature/tabs/src/main/java/mozilla/components/feature/tabs/tabstray/TabsTrayPresenter.kt

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.tabs.tabstray
+
+import android.support.v7.util.DiffUtil
+import android.support.v7.util.ListUpdateCallback
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.tabstray.TabsTray
+
+/**
+ * Presenter implementation for a tabs tray implementation in order to update the tabs tray whenever
+ * the state of the session manager changes.
+ */
+class TabsTrayPresenter(
+    private val tabsTray: TabsTray,
+    private val sessionManager: SessionManager
+) : SessionManager.Observer {
+    @Volatile private var sessions = sessionManager.sessions
```

Comment:
Why do these need to be volatile? A presenter instance isn't shared across threads, right?